Sources/ParseLiveQuery/Client.swift

```diff
@@ -149,13 +150,13 @@ extension Client {
             )
             subscriptions.append(subscriptionRecord)
 
-            if socket == nil {
-                if !disconnected {
-                    reconnect()
-                }
-            } else {
-                sendOperationAsync(.Subscribe(requestId: subscriptionRecord.requestId, query: query))
-            }
+        if socket == nil || socket?.readyState == .CLOSED{
+            userDisconnected = false
```

Comment:
I am not sure this is the right place for it.
This would trigger the reconnection on any subscription, whilst the user never explicitly called `reconnect()`.
Maybe adding a private method for `reconnect()` and moving this state change to that method would be better.